### PR TITLE
Check all option combinations in CI

### PR DESCRIPTION
**Pull request by SergioGasquez**
- Adds a `--all-combinations` flag to the `check` command
- Modified the workflow to use that flag when running scheduled, and when running it manually we should be able to select if we want the flag or not
  - I couldnt test this, yet

CI test run: https://github.com/SergioGasquez/esp-generate/actions/runs/11570938398

### CODE REVIEW

File: xtask/src/main.rs
Comment (MabezDev) [inline xtask/src/main.rs:139]: What's the purpose of this filter?

Comment (SergioGasquez) [reply xtask/src/main.rs:139]: Currently, when using ble and wifi, we throw an error: https://github.com/esp-rs/esp-generate/blob/main/src/main.rs#L408. I guess its related with https://github.com/esp-rs/esp-generate/issues/17. 

We should decide if we want to close the issue, and keep it like this. Or add support to allow using both options at the same time.

Comment (MabezDev) [reply xtask/src/main.rs:139]: We can fix this later, but we should definitely fix it as they aren't really mutually exclusive.

Probably worth fixing after lands: https://github.com/esp-rs/esp-hal/pull/2301 as it should simplify initializing things

**Review by bjoernQ — APPROVED** [review]
Thanks

For the future I have two ideas

- define the options somewhere else so the xtask and main could use the same source of truth since currently we need to duplicate options (and also the xtask doesn't take those optional things into account currently)
- maybe we could have a way to allow generating into an existing directly (but only in CI with some magic ENV-var or something) and just remove the generated files between check runs and speed up checking by not having `cargo check` check a dependency if the features etc. didn change? (Not sure how feasible that is)

File: xtask/src/main.rs
Comment (jessebraham) [reply xtask/src/main.rs:139]: > We can fix this later, but we should definitely fix it as they aren't really mutually exclusive.

FYI, the TUI does not allow selecting both, this was a temporary fix to resolve #19. See [this comment](https://github.com/esp-rs/esp-generate/issues/19#issuecomment-2431121404).

**Review by jessebraham — APPROVED** [review]
Thanks!

**Pull request merged**
